### ISSUE TITLE
$ prefix to patterns when searching word under cursor doesn't work in fish shell

### ISSUE THREAD

**Issue by allhailwesttexas** [jesseleite/vim-agriculture#13]
I've just discovered this plugin to pass file options to `:Rg` and it seems really good, thanks for creating it! Whilst trying it out I've found that the `:RgRawWordUnderCursor` & co mappings don't return any results and that removing the `$` fixes the issue. I'm not very familiar with Vimscript so I don't know if it's something with my environment or a problem for everyone, but I'll open a PR either way.

**Comment by jesseleite** [comment]
Hmm, works for me with the `$`! I use this all the time. I wonder why it's not working for you 🤔

**Comment by jesseleite** [comment]
I believe this was to preserve manual backslash escaping. It's called ANSI-C quoting when you `$'some string'` (see [bash manual here](http://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html)).

Again, I wonder why this doesn't work in your environment.  What's your OS, Vim version, bash/zsh version, etc?

**Comment by allhailwesttexas** [comment]
Thanks for the quick response! My env is:

- Ubuntu 18.04 LTS
- kitty terminal
- fish shell (but I have tried in bash as well with same issue)
- neovim 0.4.4

It seems most likely that the issue is fish shell, let me try again with bash and the built-in terminal client.

**Comment by allhailwesttexas** [comment]
OK, I changed my shell with `chsh` and restarted my computer and now it does work with the `$$$`s. I will close the PR but I don't really know what a cross-shell solution would be. Will investigate when I have time.

**Comment by jesseleite** [comment]
@allhailwesttexas ah good find! If that's all that doesn't work, I'd like to find a solution for fish shell. Will keep this open and look into it. Thanks!

**Comment by ipwnponies** [comment]
@jesseleite what is the reason behind using bash ansi-c quotes, is it to escape and forward the user query to `fzf` (akin to raw strings in python/js).

https://github.com/jesseleite/vim-agriculture/blob/1095d907930fc545f88541b14e5ea9e34d63c40f/autoload/agriculture.vim#L33

I gleaned this from [`fzf-vim` docs][1] regarding ripgrep and it feels like this might be the same use case, trying to take multi-word command line arguments and present them as the query . Does vim's `shellescape` work here? That could keep it shell agnostic 👍 

```vim
function! RipgrepFzf(query, fullscreen)
  let command_fmt = 'rg --column --line-number --no-heading --color=always --smart-case -- %s || true'
  let initial_command = printf(command_fmt, shellescape(a:query))
  let reload_command = printf(command_fmt, '{q}')
  let spec = {'options': ['--phony', '--query', a:query, '--bind', 'change:reload:'.reload_command]}
  call fzf#vim#grep(initial_command, 1, fzf#vim#with_preview(spec), a:fullscreen)
endfunction
```

[1]: https://github.com/junegunn/fzf.vim#example-advanced-ripgrep-integration

**Comment by jesseleite** [comment]
@ipwnponies Oh interesting! Will look into `shellescape()`, thanks for that info!

**Comment by allhailwesttexas** [comment]
Nice!
